spinnaker_graph_front_end/examples/hello_world_untimed/hello_world_vertex.py

```diff
@@ -0,0 +1,133 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+import logging
+from spinn_utilities.overrides import overrides
+from pacman.model.graphs.machine import MachineVertex
+from pacman.model.resources import ResourceContainer, VariableSDRAM
+from spinn_front_end_common.utilities.constants import (
+    SYSTEM_BYTES_REQUIREMENT, BYTES_PER_WORD)
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.abstract_models import (
+    AbstractGeneratesDataSpecification)
+from spinn_front_end_common.interface.buffer_management.buffer_models import (
+    AbstractReceiveBuffersToHost)
+from spinn_front_end_common.interface.buffer_management import (
+    recording_utilities)
+from spinnaker_graph_front_end.utilities import SimulatorVertex
+from spinnaker_graph_front_end.utilities.data_utils import (
+    generate_system_data_region)
+import numpy
+from pacman.executor.injection_decorator import inject_items
+
+logger = logging.getLogger(__name__)
+
+
+class HelloWorldVertex(
+        SimulatorVertex, AbstractGeneratesDataSpecification,
+        AbstractReceiveBuffersToHost):
+
+    DATA_REGIONS = Enum(
+        value="DATA_REGIONS",
+        names=[('SYSTEM', 0),
+               ('PARAMS', 1),
+               ('STRING_DATA', 2)])
+
+    PARAMS_BASE_SIZE = BYTES_PER_WORD * 2
+
+    def __init__(self, n_repeats, label, constraints=None):
+        super(HelloWorldVertex, self).__init__(
+            label, "hello_world.aplx", constraints=constraints)
+
+        self._text = label
+        text_extra = len(label) % BYTES_PER_WORD
+        if text_extra != 0:
+            for _ in range(BYTES_PER_WORD - text_extra):
+                self._text += ' '
+
+        self._n_repeats = n_repeats
+
+    @property
+    @overrides(MachineVertex.resources_required)
+    def resources_required(self):
+        resources = ResourceContainer(
+            sdram=VariableSDRAM(
+                SYSTEM_BYTES_REQUIREMENT +
+                recording_utilities.get_recording_header_size(1) +
+                self.PARAMS_BASE_SIZE + len(self._text),
+                len(self._text)))
+
+        return resources
+
+    @inject_items({
+        "data_n_steps": "DataNSteps"
+    })
+    @overrides(AbstractGeneratesDataSpecification\
+               .generate_data_specification,
+               additional_arguments=["data_n_steps"])
+    def generate_data_specification(self, spec, placement, data_n_steps):
+        # Generate the system data region for simulation .c requirements
+        # Note that the time step and time scale factor are unused here
+        generate_system_data_region(
```

Comment:
this shows the fudge. machine_time_step=0, time_scale_factor=0. your now expecting vertex writers to add gubbins to their code to satisfy a invalid api for them. 

make a seperate api which HIDES this ugly from the vertex delveoper if you NEED to use the same api.

Reply:
I have made an API call to hide this.